src/interactions/slashCommands/mixins/CommandOptionWithChoices.ts

```diff
@@ -46,17 +51,43 @@ export abstract class ApplicationCommandOptionWithChoicesBase<T extends string |
 	 *
 	 * @param choices The choices to add
 	 */
-	public addChoices(choices: [name: string, value: T][]) {
+	public addChoices(choices: [name: string, value: T][]): Omit<this, 'setAutocomplete'> {
+		if (typeof this.autocomplete !== 'undefined') {
+			throw new RangeError('Autocomplete and choices are mutually exclusive to each other.');
+		}
+
 		ow(choices, `${ApplicationCommandOptionTypeNames[this.type]} choices`, choicesPredicate);
 
 		for (const [label, value] of choices) this.addChoice(label, value);
 		return this;
 	}
 
+	/**
+	 * Marks the option as autocompletable
+	 * @param autocomplete If this option should be autocompletable
+	 */
+	public setAutocomplete(autocomplete: boolean): Omit<this, 'addChoice' | 'addChoices'> {
```

Comment:
Add an overload here so that when the value is set to `false`, it returns itself, instead of incorrectly omitting `addChoice` and `addChoices`

```suggestion
	public setAutocomplete(autocomplete: false): this;
	public setAutocomplete(autocomplete: boolean): Omit<this, 'addChoice' | 'addChoices'> {
```

Reply:
Returning this will be a bit iffy if someone does `setAutocomplete(true).setAutocomplete(false)`, won't it? (they shouldn't be doing it but well.. yeah)

Reply:
Even if they do `setAutocomplete(true).setAutocomplete(false)` it's still technically valid typing. I based this on how other parts of builders worked like `required` where you can do `setRequired(true).setRequired(false)`.

Reply:
But that's not the same at all. In those cases, there are no omitted functions, in this case there are so they might stay as missing

Reply:
Ok, I've updated the overload to return `Omit<this, 'setAutocomplete'>`

Reply:
That is absolutely not the solution. You can still use it after to set it to true.

Reply:
I'm confused, that's what my original suggestion did, no? It returned `this` if you set `autocomplete` to `false`. Which means it allowed you to still set it to `true` after setting it to `false`.

Reply:
Oh ignore me, I see what you're saying now.